test/system/post_index_spec.rb

```diff
@@ -0,0 +1,67 @@
+require 'rails_helper'
+
+RSpec.describe 'Post index page', type: :feature do
+  user = User.create(name: 'Abel', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.')
+  commenter = User.create(name: 'Salamon', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Ghana.')
+  first_post = Post.create(author_id: user.id, title: 'Hello', text: 'This is my first post')
+
+  Comment.create(user_id: commenter.id, post_id: first_post.id, text: 'comment')
+  Comment.create(user_id: commenter.id, post_id: first_post.id, text: 'another comment')
+
+  before :each do
+    visit user_posts_path(user)
+  end
+  describe "page's content" do
+    it "should see the user's profile image" do
+      expect(page).to have_css("img[src='#{user.photo}']")
+    end
+
+    it "should see the user's name" do
+      expect(page).to have_content user.name
+    end
+
+    it 'should see the number of posts' do
+      expect(page).to have_content("Number of posts: #{user.posts.count}")
+    end
+
+    it "should see the post's title" do
+      user.posts.each do |post|
+        expect(page).to have_content post.title
+      end
+    end
+
+    it "should see the post's body" do
+      user.posts.each do |post|
+        expect(page).to have_content post.text
+      end
+    end
+
+    it 'should see first comments on a post' do
+      user.posts.each do |post|
+        last_5_comments = post.find_5_comments
+        last_5_comments.each do |comment|
+          expect(page).to have_content("#{comment.author.name}: #{comment.text}")
```

Comment:
- [ ] This test and some others are failing. Kindly fix them.

  ![image](https://github.com/abel-tefera/ror-blog-app/assets/81468675/6edb04f1-24db-41aa-8235-bd5c320da3a7)